src/plugins/kubeconfig/types.ts

```diff
@@ -10,34 +10,21 @@ You should have received a copy of the GNU General Public License along with @p0
 **/
 import { PermissionSpec } from "../../types/request";
 
-export type K8sConfig = {
-  workflows: {
-    items: K8sClusterConfig[];
-  };
-};
-
 export type K8sClusterConfig = {
-  clusterId: string;
+  label?: string;
   clusterServer: string;
   clusterCertificate: string;
-  state: string;
-  awsAccountId?: string;
-  awsClusterArn?: string;
-} & (KubernetesProxyComponentConfig | KubernetesPublicComponentConfig);
-
-// These are optional in general for k8s clusters, but required for EKS
-export type EksClusterConfig = K8sClusterConfig & {
-  awsAccountId: string;
-  awsClusterArn: string;
-};
-
-type KubernetesProxyComponentConfig = {
-  isProxy: true;
+  isProxy: boolean;
+  token: string;
   publicJwk: string;
```

Comment:
`publicJwk` is not required if `isProxy` is false; if we're not discriminating on `isProxy`, should this be optional?

Reply:
we could mark it as optional.